tools/lib/pretty_print.py

```diff
@@ -37,36 +37,56 @@ def pretty_print_html(html: str, num_spaces: int = 4) -> str:
     # we proceed, we will push/pop info dictionaries on/off a stack.
     for token in tokens:
 
-        if token.kind in ('html_start', 'handlebars_start', 'handlebars_singleton',
-                          'html_singleton', 'django_start',
-                          'jinja2_whitespace_stripped_type2_start',
-                          'jinja2_whitespace_stripped_start') and stack[-1]['tag'] != 'pre':
+        if (
+            token.kind
+            in (
+                "html_start",
+                "handlebars_start",
+                "handlebars_singleton",
+                "html_singleton",
+                "django_start",
+                "jinja2_whitespace_stripped_type2_start",
+                "jinja2_whitespace_stripped_start",
+            )
+            and stack[-1]["tag"] != "pre"
+        ):
```

Comment:
This and the block below badly need extracting  constants with names like `django_start_tokens`.